pkgs/applications/misc/waylock/default.nix

```diff
@@ -30,16 +31,9 @@ stdenv.mkDerivation rec {
   ];
 
   dontConfigure = true;
+  strictDeps = false;
 
-  preBuild = ''
-    export HOME=$TMPDIR
```

Comment:
As andrewrk said in https://github.com/NixOS/nixpkgs/pull/242153#discussion_r1256635587, use ZIG_GLOBAL_CACHE_DIR instead of HOME.